src/NHibernate/Impl/StatelessSessionImpl.cs

```diff
@@ -379,8 +371,10 @@ public override string BestGuessEntityName(object entity)
 
 		public override string GuessEntityName(object entity)
 		{
-			CheckAndUpdateSessionStatus();
-			return entity.GetType().FullName;
+			using (BeginContext())
```

Comment:
Here that is the other way round: switching to `BeginContext` for matching the method on `SessionImpl`, and since what it does does not rely at all on the session state.